jupyterhub_moss/spawner.py

```diff
@@ -213,12 +214,20 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
                 raise RuntimeError("GPU(s) not available for this partition")
             options["gres"] = gpu_gres_template.format(options["ngpus"])
 
+        partition_environments = tuple(
+            self.partitions[partition]["jupyter_environments"].values()
+        )
         if "environment_path" not in options:
             # Set path to use from first environment for the current partition
-            default_venv = tuple(
-                self.partitions[partition]["jupyter_environments"].values()
-            )[0]
-            options["environment_path"] = default_venv["path"]
+            options["environment_path"] = partition_environments[0]["path"]
+
+        # Add environment_path to PATH unless it is in the settings with add_to_path=False
+        corresponding_env = find(
+            lambda env: env["path"] == options["environment_path"],
+            partition_environments,
+        )
+        if corresponding_env is None or corresponding_env.get("add_to_path", True):
```

Comment:
Given you set  a default value for the trait `add_to_path`, is the `get(..., True)` needed ?

Reply:
I add an issue with that, I didn't investigate where it comes from.

Reply:
I think `default_value` does not work in this case because it specifies a value inside a `dict` rather than a `trait`.